plaid/investment_transactions.go

```diff
@@ -0,0 +1,99 @@
+package plaid
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+type InvestmentTransaction struct {
+	InvestmentTransactionID string `json:"investment_transaction_id"`
+	AccountID               string `json:"account_id"`
+	SecurityID              string `json:"security_id"`
+	CancelTransactionID     string `json:"cancel_transaction_id"`
+
+	Date                   string  `json:"date"`
+	Name                   string  `json:"name"`
+	Quantity               float64 `json:"quantity"`
+	Amount                 float64 `json:"amount"`
+	Price                  float64 `json:"price"`
+	Fees                   float64 `json:"fees"`
+	Class                  string  `json:"class"`
+	ISOCurrencyCode        string  `json:"iso_currency_code"`
+	UnofficialCurrencyCode string  `json:"unofficial_currency_code"`
+}
+
+type GetInvestmentTransactionsResponse struct {
+	APIResponse
+	Item                        Item                    `json:"item"`
+	Accounts                    []Account               `json:"accounts"`
+	InvestmentTransactions      []InvestmentTransaction `json:"investment_transactions"`
+	Securities                  []Security              `json:"securities"`
+	TotalInvestmentTransactions int                     `json:"total_investment_transactions"`
+}
+
+type GetInvestmentTransactionsOptions struct {
+	StartDate  string
+	EndDate    string
+	AccountIDs []string
+	Count      int
+	Offset     int
+}
+
+type getInvestmentTransactionsRequest struct {
+	ClientID    string                                  `json:"client_id"`
+	Secret      string                                  `json:"secret"`
+	AccessToken string                                  `json:"access_token"`
+	StartDate   string                                  `json:"start_date"`
+	EndDate     string                                  `json:"end_date"`
+	Options     getInvestmentTransactionsRequestOptions `json:"options,omitempty"`
+}
+
+type getInvestmentTransactionsRequestOptions struct {
+	AccountIDs []string `json:"account_ids"`
+	Count      int      `json:"count"`
+	Offset     int      `json:"offset"`
+}
+
+// GetInvestmentTransactionsWithOptions retrieves user-authorized investment transaction data for investment-type accounts.
+// See https://plaid.com/docs/api/#investment-transactions.
+func (c *Client) GetInvestmentTransactionsWithOptions(accessToken string, options GetInvestmentTransactionsOptions) (resp GetInvestmentTransactionsResponse, err error) {
+	if options.StartDate == "" || options.EndDate == "" {
```

Comment:
For good measure can we check that `accessToken  != ""` - similar to `GetHoldings`?

Reply:
⚡️